app/partials/subject-uploader.cjsx

```diff
@@ -75,19 +84,21 @@ module.exports = React.createClass
       @finish()
 
   finish: ->
-    newSubjectIDs = (id for {id} in @state.batch)
-    @props.subjectSet.addLink 'subjects', newSubjectIDs
-      .then =>
-        @state.batch.splice 0
-
-        if @state.current is @props.subjects.length
-          @props.onComplete
-            successes: @state.successes
-            errors: @state.errors
-
-      .catch (error) =>
-        console.error 'TODO: handle linking error', error
-
-      .then =>
-        if @isMounted()
-          @setState inProgress: false
+    unless @state.batch is 0
```

Comment:
Maybe `@state.batch.length ` here? It looks like `@state.batch` is always an array